conda/resolve.py

```diff
@@ -544,26 +572,14 @@ def filter_group(_specs):
         # perfect about this, so performance matters.
         for _ in range(2):
             snames.clear()
-            slist = list(explicit_specs)
+            slist = deque(explicit_specs)
             reduced = False
             while slist:
-                s = slist.pop()
+                s = slist.popleft()
                 top_level_spec = s
                 reduced = filter_group([s])
                 if reduced:
                     slist.append(s)
-                elif reduced is None:
```

Comment:
Removing this early bail-out allows get_reduced_index to serve as a satisfiability check.  Any entries with 0 results are part of a conflict.  This avoids a SAT call, at the cost of doing more filtering in get_reduced_index.  I'm not sure which will be faster in practice.  Based on the test suite, they seem about the same, but this get_reduced_index approach may have more room for optimization / conversion to compiled code.